apps/frontpage/app/sitemap-all/docs/sitemap.ts

```diff
@@ -0,0 +1,52 @@
+import { MetadataRoute } from 'next';
+import { generateDocsTree } from '../../../lib/get-tree';
+import { TreeProps, docsVersions } from '@repo/utils';
+
+export default function sitemap(): MetadataRoute.Sitemap {
+  const latestVersion = docsVersions[0];
+  const listofTrees = docsVersions.map((version) => {
+    return {
+      version: version.id,
+      tree: generateDocsTree(`content/docs/${version.id}`),
+    };
+  });
+
+  const flattenedTree: TreeProps[] = [];
+
+  const flattenTree = (tree: TreeProps[]) => {
+    tree.forEach((node) => {
+      if (node.children) {
+        flattenedTree.push(node);
+        flattenTree(node.children);
+      }
+      flattenedTree.push(node);
+    });
+  };
+
+  listofTrees.forEach((list) => {
+    flattenTree(list.tree);
+  });
+
+  const filteredTree = flattenedTree
+    .filter((node) => node.type !== 'directory')
+    .filter((node) => node.draft !== true);
+
+  const docs = filteredTree.map((node) => {
+    const url = `https://storybook.js.org${node.slug}`
+      .replace(`${latestVersion.id}/`, '')
+      .replace(docsVersions[1].id, docsVersions[1].inSlug || '')
+      .replace(docsVersions[2].id, docsVersions[2].inSlug || '')
```

Comment:
We don't currently have one, but we _can_ have a pre-release version, which would mean there's a `docsVersions[3]`.

It makes me nervous to have hard-coded array lengths like this instead of mapping over an array that could be any length.

Reply:
The real refactor that need to happen is to make sure the slug reflect the real scenario so we don't have to do any modifications here.

Reply:
You're right. I forgot that we had a `getUrl` function that we made for the sidebar. I made it globally available so we can use it here as well.